### PR TITLE
Fix für #579

### LINKED ISSUE #579: Umleitungsfehler bei selbst erstellten Seiten Layout Wechsel
Wenn man auf einer Selbst erstellten Seite (im Frontend) befindet und dann das Layout wechselt, kommt ein Umleitungsfehler.

### PR DESCRIPTION
**Pull request by hhunderter**
# Description

Fix für #579
Fixes #579 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# This PR has been tested in the following browsers:

- [ ] Chrome
- [x] Firefox
- [ ] Opera
- [ ] Edge
- [ ] IE

### CODE REVIEW

File: application/libraries/Ilch/Request.php [hunk 1]
```diff
@@ -224,19 +237,36 @@ public function getParam($key)
...
-    public function setParam($key, $value)
+    public function setParam(string $key, ?string $value): Request
     {
         $this->params[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets the param with the given key / value.
```
Comment (blackcoder87) [inline application/libraries/Ilch/Request.php:251]: Wrong description for this function.

File: application/libraries/Ilch/Request.php [hunk 2]
```diff
@@ -326,4 +358,47 @@ public function isSecure()
 
         return $returnValue;
     }
+
+    /**
```
Comment (blackcoder87) [inline application/libraries/Ilch/Request.php:362]: Could use some description of what array. array of url parts..

File: application/modules/admin/plugins/AfterDatabaseLoad.php [hunk 3]
```diff
@@ -19,20 +19,32 @@ public function __construct(array $pluginData)
...
+                unset($urlParts[1]);
+            }
+
+            $result = $router->convertParamStringIntoArray(implode('/', $urlParts));
+            foreach ($result as $key => $value) {
+                $request->setParam($key, $value);
+            }
         }
 
         // Log the entrys
```
Comment (blackcoder87) [inline application/modules/admin/plugins/AfterDatabaseLoad.php:43]: Not your change, but this should be "entries" instead of "entrys".

**Review by blackcoder87 — APPROVED** [review]

**Comment by hhunderter** [comment]
wollte eig nachher auch eine Änderung hochladen ;)

**Pull request merged**